### PR TITLE
Add examples section

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #82

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,52 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+  `for-all-boosters.sh -p cmd "pwd"`
+    
+  The key flag here is `-p`, which forces the script to create a boosters locally. `cmd "pwd"` Makes the script execute the `pwd` command for each booster - effectively working as a noop
+  
+* Execute a simple version increment a on selected number of boosters for the `master` branch
+
+  `./for-all-boosters.sh -b master -m cache,http change_version`    
+
+  The `-b` flag ensures that only the `master` branch will be used while the `-m` flag ensures that only the `cache` and `http` boosters are used
+  The `change_version` command will simply increment the SB version, for example from `1.5.10-SNAPSHOT` to ``1.5.11-SNAPSHOT``
```
Comment (metacosm) [inline README.md]: This isn't correct. `change_version` will not change the SB version but change the *booster*'s version from `1.5.10-1-redhat-SNAPSHOT` to `1.5.10-2-redhat-SNAPSHOT` for example.

File: README.md [hunk 2]
```diff
@@ -15,6 +15,52 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+
+  The `-x` flag ensures that all boosters are used except `cache` and `http` boosters.
+  The `change_version` command will set the parent version (because of the `-p` flag) of the boosters to `24` 
+  
+* Execute an arbitrary command on all boosters without commit and pushing changes to remote
+
+  `./for-all-boosters.sh -d cmd "cp $(pwd)/.editorconfig . && git add .editorconfig"`
+  
+  The `-d` flag ensures that no changes are committed locally or pushed to the remote repos.
+  The arbitrary command here adds the `.editocconfig` file (which is in the same directory as `for-all-boosters.sh`) to each booster and makes git track it
```
Comment (metacosm) [inline README.md]: typo: `editocconfig` -> `editorconfig`

File: README.md [hunk 3]
```diff
@@ -15,6 +15,52 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+  `./for-all-boosters.sh -d cmd "cp $(pwd)/.editorconfig . && git add .editorconfig"`
+  
+  The `-d` flag ensures that no changes are committed locally or pushed to the remote repos.
+  The arbitrary command here adds the `.editocconfig` file (which is in the same directory as `for-all-boosters.sh`) to each booster and makes git track it   
+
+* Execute a custom script for each booster on some branch ignoring whatever changes exist locally
+
+  `./for-all-boosters.sh -f -m foo cmd "/home/scripts/adhoc.py"`
+  
+  Before running the command, the script will show a warning if local changes exist (to ignore the warning add the `-n` flag)
```
Comment (metacosm) [inline README.md]: No, ignoring the local changes is done with `-f` option. `-n` is to remove confirmation dialogs ;)

File: README.md [hunk 4]
```diff
@@ -15,6 +15,52 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+  Any changes that the `adhoc.py` script makes to the booster files will automatically be committed and pushed to the `foo` branch (which needs to exist)
```
Comment (metacosm) [inline README.md]: No, this only happens if you provide a commit message using the `-p` option of the `cmd` command.

File: README.md (hunk 3 above)
Comment (geoand) [reply README.md]: I thought that is what I said but let me take another look

File: README.md (hunk 4 above)
Comment (geoand) [reply README.md]: Thanks! I will update the example

File: README.md (hunk 1 above)
Comment (geoand) [reply README.md]: Correct of course, that's what I had in mind but got confused due to all the different versions :)

**Comment by geoand** [comment]
@metacosm Issues addressed

File: README.md [hunk 5]
```diff
@@ -15,6 +15,52 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+  `for-all-boosters.sh -p cmd "pwd"`
+    
+  The key flag here is `-p`, which forces the script to create a boosters locally. `cmd "pwd"` Makes the script execute the `pwd` command for each booster - effectively working as a noop
+  
+* Execute a simple version increment a on selected number of boosters for the `master` branch
+
+  `./for-all-boosters.sh -b redhat -m cache,http change_version`    
+
+  The `-b` flag ensures that only the `master` branch will be used while the `-m` flag ensures that only the `cache` and `http` boosters are used
+  The `change_version` command will simply increment the version of each booster, for example from `1.5.10-1-redhat-SNAPSHOT` to ``1.5.10-2-redhat-SNAPSHOT``
```
Comment (metacosm) [inline README.md]: Maybe we should clarify that the new version will be automatically computed for each booster, not a version that will apply uniformly to all boosters?

Comment (geoand) [reply README.md]: Good idea

Comment (geoand) [reply README.md]: Fixed

File: README.md [hunk 6]
```diff
@@ -15,6 +15,54 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+  
+  The `-d` flag ensures that no changes are committed locally or pushed to the remote repos.
+  The arbitrary command here adds the `.editorconfig` file (which is in the same directory as `for-all-boosters.sh`) to each booster and makes git track it   
+
+* Execute a custom script for each booster on some branch ignoring whatever changes exist locally
+
+  `./for-all-boosters.sh -f -m foo cmd -p "Made some change" "/home/scripts/adhoc.py"`
+  
+  Due to the presense of the `-f` flag, before running the command the script will show a warning if local changes exist (to remove the warning add the `-n` flag)
+  Any changes that the `adhoc.py` script makes to the booster files will automatically be committed using the message specified in `-p` and pushed to the `foo` branch (which needs to exist)
```
Comment (metacosm) [inline README.md]: If the idea is to push to the `foo` branch, then the `-b` option should have been used. You used `-m` meaning only operate on the `spring-boot-foo-booster` ;)

Comment (geoand) [reply README.md]: Shoot... I always confuse those

Comment (geoand) [reply README.md]: Fixed

File: README.md [hunk 7]
```diff
@@ -15,6 +15,54 @@ Run `for-all-boosters.sh -h` for an overview of what the script can do and how t
...
+
+  `for-all-boosters.sh -p cmd "pwd"`
+    
+  The key flag here is `-p`, which forces the script to create a boosters locally. `cmd "pwd"` Makes the script execute the `pwd` command for each booster - effectively working as a noop
+  
+* Execute a simple version increment a on selected number of boosters for the `master` branch
+
+  `./for-all-boosters.sh -b redhat -m cache,http change_version`    
+
+  The `-b` flag ensures that only the `master` branch will be used while the `-m` flag ensures that only the `cache` and `http` boosters are used
```
Comment (metacosm) [inline README.md]: I missed that, but the command references the `redhat` branch, not `master`.

Comment (geoand) [reply README.md]: Good catch

**Comment by geoand** [comment]
Thanks @metacosm for your patience :)

**Pull request merged**